### PR TITLE
Removing special declaration of instances with `Context` outside sections in `Module Type`

### PR DESCRIPTION
**Pull request by herbelin**
So as to hopefully simplify the implementation, the PR experiments removing the special treatment of instances with `Context` outside sections in `Module Type`.
    
This means:
- Context outside section does not declare defined _instances_ anymore in Module Type.
- Context outside section does not declare axiomatic _instances_ anymore with a local flag in Module Type.
    
Note: Declarations themselves made by Context outside sections in `Module Type` remain exceptionally "Global" while they are "Local" otherwise (not visible w/o qualifications). See #10888. It seems difficult to change this behavior in one way or the other.

- [x] Added / updated **test-suite**.
- [x] Added **changelog**.

### CODE REVIEW

**Comment by herbelin** [comment]
@coqbot run full ci

**Comment by herbelin** [comment]
This PR removes a corner case in declaration of instances. It is not used in CI and it would help to simplify the implementation to remove it (see #13445).

File: doc/changelog/02-specification-language/18254-master+def-in-context-instance-in-modtype.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+- **Changed:**
+  Definitions done using :cmd:`Context` outside a section in a
+  :cmd:`Module Type` do not any more try to declare a class
+  instance. Assumptions bound to a class and declared using
```
Comment (SkySkimmer) [inline doc/changelog/02-specification-language/18254-master+def-in-context-instance-in-modtype.rst:4]: "bound to a class" sounds wrong, it's more "whose type is a class" no?

File: doc/changelog/02-specification-language/18254-master+def-in-context-instance-in-modtype.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+- **Changed:**
+  Definitions done using :cmd:`Context` outside a section in a
```
Comment (SkySkimmer) [inline doc/changelog/02-specification-language/18254-master+def-in-context-instance-in-modtype.rst:2]: Maybe we could emphasize that this is about definitions (ie with `:= body`) vs assumptions, I got confused for a bit.

Comment (herbelin) [reply doc/changelog/02-specification-language/18254-master+def-in-context-instance-in-modtype.rst:2]: You mean something like that:
```suggestion
  Declarations of the form `id := body` in :cmd:`Context` outside a section in a
```

File: doc/changelog/02-specification-language/18254-master+def-in-context-instance-in-modtype.rst (hunk 1 above)
Comment (herbelin) [reply doc/changelog/02-specification-language/18254-master+def-in-context-instance-in-modtype.rst:4]: I was not satisfied either. Your suggestion is ok to me.

File: doc/changelog/02-specification-language/18254-master+def-in-context-instance-in-modtype.rst (hunk 2 above)
Comment (SkySkimmer) [reply doc/changelog/02-specification-language/18254-master+def-in-context-instance-in-modtype.rst:2]: yes

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by proux01** [comment]
@SkySkimmer CI green
